src/librustc_infer/infer/region_constraints/mod.rs

```diff
@@ -137,14 +137,23 @@ pub enum Constraint<'tcx> {
     RegSubReg(Region<'tcx>, Region<'tcx>),
 }
 
-impl Constraint<'_> {
+impl<'tcx> Constraint<'tcx> {
     pub fn involves_placeholders(&self) -> bool {
         match self {
             Constraint::VarSubVar(_, _) => false,
             Constraint::VarSubReg(_, r) | Constraint::RegSubVar(r, _) => r.is_placeholder(),
             Constraint::RegSubReg(r, s) => r.is_placeholder() || s.is_placeholder(),
         }
     }
+
+    pub fn to_region_outlives_predicate(self) -> OutlivesPredicate<RegionKind, RegionKind> {
+        match self {
+            Self::VarSubVar(a, b) => OutlivesPredicate(RegionKind::ReVar(a), RegionKind::ReVar(b)),
```

Comment:
`RegionOutlivesPredicate<'tcx> = OutlivesPredicate<Region<'tcx>, Region<'tcx>>`, but I got `OutlivesPredicate<RegionKind<'tcx>>` here, struck by how to convert it into `RegionOutlivesPredicate<'tcx>`(cannot return [Region](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/type.Region.html) which refers to a local RegionKind.